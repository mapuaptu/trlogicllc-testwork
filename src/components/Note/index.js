import Icon from '@/components/Icon/index.vue';

export default {
  name: 'Note',
  components: {
    Icon,
  },
  props: {
    title: {
      type: String,
      default: '',
    },
  },
};
