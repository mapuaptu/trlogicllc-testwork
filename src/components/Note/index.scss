@import '@/styles/_variables.scss';

.note {
  position: relative;
  left: 0;
  border: 1px solid $color-java;
  border-radius: 5px;
  transition: $transition;

  .title {
    display: flex;
    align-items: center;
    justify-content: space-between;
    border-bottom: 1px solid $color-java;
    padding: 10px 20px;
    font-weight: 700;
    font-size: 20px;
    color: $color-white;
    background-color: $color-java;

    .icon {
      transition: $transition;
      cursor: pointer;

      &:hover {
        opacity: 0.6;
      }
    }
  }

  .controls {
    display: flex;

    > div:not(:last-child) {
      margin-right: 15px;
    }
  }

  .content {
    padding: 10px 20px;
    color: $color-white;
  }

  &:not(:last-child) {
    margin-bottom: 25px;
  }

  &:hover {
    left: 5px;
  }
}
