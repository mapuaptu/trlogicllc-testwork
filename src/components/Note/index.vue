<template>
  <div :class="$style.note">
    <div :class="$style.title">
      <div>{{ title }}</div>

      <Icon
        icon="delete-outline"
        :class="$style.icon"
        @click.native="$emit('delete')"
      />
    </div>

    <div :class="$style.content">
      <slot />
    </div>
  </div>
</template>

<script src="./index.js"></script>
<style src="./index.scss" lang="scss" module></style>
