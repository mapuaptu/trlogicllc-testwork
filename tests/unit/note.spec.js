import Note from '@/components/Note/index.vue';
import { mount } from '@vue/test-utils';

describe('Note component', () => {
  it('1. Rendered success', async () => {
    const wrapper = await mount(Note);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('2. Render title success', async () => {
    const testTitle = 'testTitle';

    const wrapper = await mount(Note, {
      propsData: {
        title: testTitle,
      },
    });

    const title = wrapper.find('[data-test="note-title"]');

    expect(title.text()).toEqual(testTitle);
  });
});
