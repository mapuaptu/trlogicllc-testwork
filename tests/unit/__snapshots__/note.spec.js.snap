// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note component 1. Rendered success 1`] = `
<div>
  <div>
    <div data-test="note-title">

    </div>
    <div>
      <div><svg width="24" height="24" viewbox="0 0 24 24">
          <path fill="#fff"></path>
        </svg></div>
      <div><svg width="24" height="24" viewbox="0 0 24 24">
          <path d="M6,19A2,2 0 0,0 8,21H16A2,2 0 0,0 18,19V7H6V19M8,9H16V19H8V9M15.5,4L14.5,3H9.5L8.5,4H5V6H19V4H15.5Z" fill="#fff"></path>
        </svg></div>
    </div>
  </div>
  <div></div>
</div>
`;
